packages/babel-generator/src/generators/types.js

```diff
@@ -120,8 +120,11 @@ export function NullLiteral() {
 
 export function NumericLiteral(node: Object) {
   const raw = this.getPossibleRaw(node);
+  const opts = this.format.jsescOption;
   const value = node.value + "";
-  if (raw == null) {
+  if (opts.numbers) {
+    this.number(jsesc(value, opts));
```

Comment:
`value` is a string, but `jsesc` [expects a number](https://github.com/mathiasbynens/jsesc#numbers). You can use `node.value`.